New Adapter: Kueez

saar120 opened the pull request:
https://github.com/prebid/prebid.github.io/pull/5814

przemkaczmarek commented on static/bidder-info/kueezrtb.yaml:
```diff
@@ -0,0 +1,20 @@
+endpoint: "https://prebidsrvr.kueezrtb.com/openrtb/"
+maintainer:
+  email: "rtb@kueez.com"
+gvlVendorID: 1165
```
``` "1165": {
      "id": 1165,
      "name": "Kueez Entertainment Ltd.",
      "purposes": [1, 2, 3, 4],
      "legIntPurposes": [7, 10],
      "flexiblePurposes": [2],
      "specialPurposes": [1, 2],
      "features": [1, 2, 3],
      "specialFeatures": [],
      "cookieMaxAgeSeconds": 2592000,
      "usesCookies": true,
      "cookieRefresh": true,
      "urls": [
        {
          "langId": "en",
          "privacy": "https://weare.kueez.com/legal-center/privacy-policy",
          "legIntClaim": "https://weare.kueez.com/legal-center/privacy-policy"
        }
      ],
      "usesNonCookieAccess": true,
      "dataRetention": {
        "stdRetention": 30,
        "purposes": {

        },
        "specialPurposes": {

        }
      },
      "dataDeclaration": [1, 2, 3, 4, 6, 8, 11],
      "deviceStorageDisclosureUrl": "https://en.kueez.com/tcf.json"
    }, ```

przemkaczmarek commented on static/bidder-info/kueezrtb.yaml:
```diff
@@ -0,0 +1,20 @@
+endpoint: "https://prebidsrvr.kueezrtb.com/openrtb/"
+maintainer:
+  email: "rtb@kueez.com"
```
I sent a mail. Please confirm

assa-kariv replied:
confirmed

przemkaczmarek commented on static/bidder-info/kueezrtb.yaml:
```diff
@@ -0,0 +1,20 @@
+userSync:
+  iframe:
+    url: https://sync.kueezrtb.com/api/user/html/62ce79e7dd15099534ae5e04?pbs=true&gdpr={{.GDPR}}&gdpr_consent={{.GDPRConsent}}&us_privacy={{.USPrivacy}}&redirect={{.RedirectURL}}&gpp={{.GPP}}&gpp_sid={{.GPPSID}}
+    userMacro: ${userId}
```
it works
![image](https://github.com/user-attachments/assets/75269297-a1cc-4c02-b6e4-437cecead101)

saar120 commented on static/bidder-info/kueezrtb.yaml:
```diff
@@ -0,0 +1,20 @@
+endpoint: "https://prebidsrvr.kueezrtb.com/openrtb/"
```
@przemkaczmarek 
You need to add the `cId` to the request

https://github.com/prebid/prebid-server/blob/bf7ab4288afb0332963eb9adf081517775cdf040/adapters/kueezrtb/kueezrtb.go#L54-L62

![CleanShot 2024-11-19 at 11 51 03](https://github.com/user-attachments/assets/94b5e6c3-f4b6-4c39-a81c-2f78a1cca52b)

przemkaczmarek commented on adapters/kueezrtb/kueezrtb.go:
```diff
@@ -0,0 +1,134 @@
+	return requests, errors
+}
+
+func getMediaTypeForBid(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
```
Please put this func at end of adapter

przemkaczmarek commented on adapters/kueezrtb/params_test.go:
```diff
@@ -0,0 +1,42 @@
+}
+
+var invalidParams = []string{
+	`{"cId": 3898334}`,
```
what happend if You add:
var invalidParams = []string{
	`{}`,
	`{"cId": 123}`,
	`{"cId": true}`,
	`{"cId": null}`,
	`{"cId": ""}`,
	`{"cId": "   "}`,
	`{"cId": "cid_with_invalid_chars!@#"}`,
	`{"cId": "valid_cid", "extra_field": "unexpected"}`,
	`{"cId": "a_very_long_connection_id_exceeding_the_maximum_allowed_length_..."}`,
	`[]`,
	`["cId"]`,
	`"string_value"`,
	`42`,
	`true`,
	`{"cId": "valid_cid"`,
	`{"cId":}`,
	`{"cid": "valid_cid"}`,
}
please add more invalidParams

saar120 replied:
Added more..

bsardo commented:
Hi @saar120, we noticed that your adapter is basically the same as the Vidazoo adapter. We would like to point out that aliasing an adapter may be an option for you and is preferable if you and another bidder share the same server. You can read more about aliasing here: [aliasing an adapter](https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html#aliasing-an-adapterhttps://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html#aliasing-an-adapter).

Please let us know your thoughts on whether this is a possibility for you.

saar120 commented:
Hey @bsardo 
We are providing the adapter and the server as a white label solution.
KueezRtb and Vidazoo are different entities and should not share the same adapter.
For now this solution does not fit our needs.

przemkaczmarek commented:
@assa-kariv In our policy, we require two approvals.

saar120 commented:
@bsardo

assa-kariv commented:
@bsardo @przemkaczmarek please assist here

VeronikaSolovei9 commented:
New adapters must also open a PR against the Prebid Docs repository here: https://github.com/prebid/prebid.github.io/

Please see the [User Documentation](https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html#user-documentation) section of our new adapter developer guide.

VeronikaSolovei9 commented on adapters/kueezrtb/kueezrtb.go:
```diff
@@ -0,0 +1,134 @@
+	if err := jsonutil.Unmarshal(bidderExt.Bidder, &impExt); err != nil {
+		return "", fmt.Errorf("unmarshal ImpExtkueez: %w", err)
+	}
+	return strings.TrimSpace(impExt.ConnectionId), nil
```
Nitpick: `strings.TrimSpace` is not needed here, because "pattern": "^[a-zA-Z0-9_]+$" in kueezrtb.yaml already checks there are no spaces in the cId.

VeronikaSolovei9 reviewed (commented):
Thank you for the contribution! Code looks good to me. I added a comment about optional optimization. Please also  add a documentation PR. 
Thank you!

saar120 commented:
@VeronikaSolovei9 @przemkaczmarek 
Thank you.
Added in the docs that we support pbs (PR link in the description)
also implemented the suggested improvement

VeronikaSolovei9 reviewed (approved):
Thank you for the changes! Approved
